File: dart/03-operators/operators_arithmetic.dart
void main() {
  int a = 17;
  int b = 5;

  print('a + b = ${a + b}');   // Addition
  print('a - b = ${a - b}');   // Subtraction
  print('a * b = ${a * b}');   // Multiplication
  print('a / b = ${a / b}');   // Division (returns double)
  print('a ~/ b = ${a ~/ b}'); // Integer division
  print('a % b = ${a % b}');   // Modulo (remainder)

  // Unary operators
  int x = 10;
  print('-x = ${-x}'); // Negation

  // Increment and decrement
  int counter = 0;
  counter++;           // Post-increment
  ++counter;           // Pre-increment
  print('counter = $counter');
}

File: matlab/03-operators/operators_arithmetic.m
% Arithmetic operators in MATLAB
a = 10;
b = 3;

fprintf('a + b  = %d\n', a + b);    % Addition
fprintf('a - b  = %d\n', a - b);    % Subtraction
fprintf('a * b  = %d\n', a * b);    % Multiplication
fprintf('a / b  = %.4f\n', a / b);  % Right division
fprintf('a \\ b  = %.4f\n', a \ b); % Left division (b / a)
fprintf('a ^ b  = %d\n', a ^ b);    % Exponentiation
fprintf('mod    = %d\n', mod(a, b));% Modulo (function, not operator)

% Negation (unary minus)
disp(-a)
